main.go

```diff
@@ -134,10 +148,15 @@ func initHttpProf(logger *zap.Logger) {
 }
 
 func initMetricsServer(logger *zap.Logger) {
-	// If -dry-run=true is set, service stops here after validation
 	if dryRun {
-		os.Exit(0)
+		return
+	}
+
+	hostname, found := os.LookupEnv("HOSTNAME")
+	if !found {
+		hostname = defaultInstanceName
```

Comment:
You can also try with [`os.Hostname()`](https://pkg.go.dev/os#Hostname).

Reply:
Sure, changed

Reply:
I meant that you could use both methods :slightly_smiling_face: the `HOSTNAME` environment variable, and `os.Hostname()`.
Consider creating a function to obtain the hostname from one of these two methods, plus the fallback of using `localhost`.